asllib/Typing.ml

```diff
@@ -3767,6 +3780,7 @@ module Annotate (C : ANNOTATE_CONFIG) : S = struct
             match keyword with
             | GDK_Config -> t
             | _ ->
+                let+ () = check_no_precision_loss ~loc t_e in
```

Comment:
This seems out of place compared to the local version - if the inferred type for the expression type-satisfies the declared one, we should not error even if there was a loss of precision. For example, this global declaration should not produce an error but I think it does (even though the local version does not):
```
let y : integer{0..2^128} = UInt(Zeros{64}) * UInt(Zeros{64});
```